app/controllers/agent/ManageAgentController.scala

```diff
@@ -0,0 +1,131 @@
+import play.api.Logger
+import play.api.data.{Form, Forms}
+import play.api.i18n.MessagesApi
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Request}
+import services.AgentRelationshipService
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.propertylinking.errorhandler.CustomErrorHandler
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
+
+class ManageAgentController @Inject()(
+      val errorHandler: CustomErrorHandler,
+      authenticated: AuthenticatedAction,
+      agentRelationshipService: AgentRelationshipService,
+      manageAgentPage: views.html.propertyrepresentation.manage.manageAgent,
+      manageAgentZeroPropertyLinks: views.html.propertyrepresentation.manage.manageAgentZeroPropertyLinks,
+      manageAgentOnePropertyLinkWithAgentAssigned: views.html.propertyrepresentation.manage.manageAgentOnePropertyLinkWithAgentAssigned)(
+      implicit override val messagesApi: MessagesApi,
+      override val controllerComponents: MessagesControllerComponents,
+      executionContext: ExecutionContext,
+      val config: ApplicationConfig
+) extends PropertyLinkingController {
+
+  val logger = Logger(this.getClass.getName)
+
+  def manageAgent(agentCode: Option[Long]): Action[AnyContent] = authenticated.async { implicit request =>
+    getManageAgentPage(agentCode).map { pageOpt =>
+      pageOpt match {
+        case None       => NotFound(errorHandler.notFoundTemplate)
+        case Some(page) => Ok(page)
+      }
+    }
+  }
+
+  private[agent] def getManageAgentPage(
+        agentCode: Option[Long],
+        submitManageAgentForm: Form[ManageAgentRequest] = ManageAgentRequest.submitManageAgentRequest)(
+        implicit request: Request[_]) =
+    for {
+      organisationsAgents <- agentRelationshipService.getMyOrganisationAgents()
+      agentToBeManagedOpt = agentCode match {
+        case None if organisationsAgents.agents.size == 1 => Some(organisationsAgents.agents.head)
+        case Some(code)                                   => organisationsAgents.agents.find(a => a.representativeCode == code).map(agent => agent)
+        case _                                            => None
+      }
+      propertyLinks: OwnerAuthResult <- agentRelationshipService.getMyOrganisationsPropertyLinks(
+                                         searchParams = GetPropertyLinksParameters(),
+                                         pagination = PaginationParams(
+                                           startPoint = 1,
+                                           pageSize = 1000,
+                                           requestTotalRowCount = false),
+                                         representationStatusFilter =
+                                           Seq(RepresentationApproved, RepresentationPending)
+                                       )
+    } yield {
+      (propertyLinks.total, agentToBeManagedOpt) match {
```

Comment:
I dont fully understand what this section is doing, could you maybe explain the logic here.

Reply:
I will add comments to make it clearer. Each case is basically a scenario in the story